vm-setup/roles/v1aX_integration_test/tasks/verify.yml

```diff
@@ -84,23 +97,27 @@
 
   # Check for pods & nodes on the target cluster
   - name: Check if pods in running state
-    shell: "kubectl get pods --kubeconfig /tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml -A -o json | jq -r '.items[].status.phase' | grep -v Running"
+    k8s_info:
+      api_version: v1
+      kind: Pod
+      namespace: "{{ NAMESPACE }}"
+      kubeconfig: /tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml
+      field_selectors:
+        - status.phase=Running
     retries: 150
     delay: 20
     register: running_pods
-    failed_when: >
-      (running_pods.stderr != "") or
-      (running_pods.rc > 1) or
-      (running_pods.stdout != "")
-    until: running_pods.stdout == ""
+    until: running_pods.resources | length == 0
```

Comment:
`grep -v` is an inverted match, so in the shell we are listing the pods that are _not_ in running phase and waiting until the output is empty. However, I think the field selector with the until clause here is doing the opposite? If so, I expect we have to fix this.

These "check" task names could be clearer as well to indicate that we're waiting for a condition.

Reply:
Oh gotcha! :facepalm: